public/app/features/dashboard/panel_editor/QueryEditorRow.tsx

```diff
@@ -284,6 +285,7 @@ export class QueryEditorRow extends PureComponent<Props, State> {
     }
 
     const editor = this.renderPluginEditor();
+    const DatasourceCheatsheet = datasource.components?.ExploreStartPage;
```

Comment:
This feels awkward, should we rename this to QueryEditorHelp? feels like the way this is implemented can't it just be part of the QueryEditor?  Feels like Grafana does not do much beyond rendering the component below the editor, something the QueryEditor component itself could easily do. Guess the expand / collapse state is nice to reuse. And making this a component like this makes it possible for Grafana to change the design for how to access it. Was thinking to make we make it an icon in the query row bar instead (next to duplicate and the eye icons).

Reply:
No feelings about the naming, but for making this into a datasource API the main thing was that before it was shown outside of the editorRow (in Explore) and I think it may be the case in the future (like showing in modal, slide panel or what ever). There are other ways to solve that (provide something like `openModal` prop for the editor) but in general this makes it easier to provide consistent look and feel for the editor/query help which I think is a good think in contrast of having it implemented/styled differently across editors (especially if talking about 3rd party data sources). At the same time this does not prevent still implementing it in the queryEditor if somebody wants to just gives an easy to opt in way to do it.

Reply:
yea, agree fully. Having this be a standard component that the plugin exposes makes it possible for Grafana to refine and change how this is presented and make it consistent. 

The awkward part was "ExploreStartPage" name for the component, proposing a more generic name like QueryEditorHelp